### PR TITLE
SDK-312 [Java SDK] write and batchWrite should return data from PoPAPI response

### PR DESCRIPTION
**Pull request by VladimirLevitskiy**
Pull Request Checklist
======================

Please read this for our PR Culture:
------------------------------------
https://incountry.atlassian.net/wiki/spaces/ED/pages/46858869/Pull+requests+culture+Draft

Work relates to:
----------------
Link(s) to Jira Task/User Story/Bug/Epic:
- [SDK-312](https://incountry.atlassian.net/browse/SDK-312)

Code Quality (Before submitting PR)
------------
- [x] Ensure code compiles and passes Sonar Lint.
- [x] Do your best to ensure integration tests pass. If they don't then...
- [x] Remove, ignore, or mark pending any brittle tests.
- [x] Remove commented out code.

Target Release
--------------
- [ ] Please put the target release vehicle this needs to be bundled with.
- [ ] Did you increment the version? What is the new version?

### CODE REVIEW

File: src/main/java/com/incountry/residence/sdk/StorageImpl.java [hunk 1]
```diff
@@ -240,8 +241,13 @@ private void checkAttachmentParameters(String country, String key, String fileId
     public Record write(String country, Record newRecord) throws StorageClientException, StorageServerException, StorageCryptoException {
         HELPER.check(StorageClientException.class, newRecord == null, MSG_ERR_NULL_RECORD);
         checkCountryAndRecordKey(country, newRecord.getRecordKey());
-        TransferRecord recordedRecord = dao.createRecord(country, transformer.getTransferRecord(newRecord));
-        return transformer.getRecord(recordedRecord);
+        TransferRecord recordedTransferRecord = dao.createRecord(country, transformer.getTransferRecord(newRecord));
+        if (recordedTransferRecord == null) {
+            return newRecord;
+        }
+        Record recorderRecord = transformer.getRecord(recordedTransferRecord);
```
Comment (RommeroInCountry) [inline src/main/java/com/incountry/residence/sdk/StorageImpl.java:248]: please see C# note on the naming

**Review by RommeroInCountry — CHANGES_REQUESTED** [review]

**Review by RommeroInCountry — DISMISSED** [review]

File: README.md [hunk 2]
```diff
@@ -665,6 +608,67 @@ FindFilter filter = new FindFilter()
...
+#### Sorting
+
+---
+**NOTE**
+
+Sorting find results is currently available for InCountry dedicated instances only. Please check your subscription plan for details. This may require specifying your dedicated instance endpoint when configuring Java SDK Storage.
+
+---
+
+By default, data at a find result is not sorted. To sort the returned records by one or multiple keys use method `sortBy` of `FindFilter` .
```
Comment (artemgolozubov) [inline README.md:620]:
```suggestion
By default, data at a find result is not sorted. To sort the returned records by one or multiple keys use method `sortBy` of `FindFilter`.
```

**Review by artemgolozubov — APPROVED** [review]
LGTM

**Review by artemgolozubov — APPROVED** [review]
LGTM

**Review by RommeroInCountry — APPROVED** [review]

**Pull request merged**